agent/src/main/java/com/microsoft/hydralab/agent/runner/scanner/ApkCanaryExecutor.java

```diff
@@ -0,0 +1,264 @@
+                throw new RuntimeException("mkdir fail!");
+            }
+        }
+
+        int code = -1;
+
+        String name = apk.getName();
+        String itemName = getAPKAnalysisItemName(name);
+        String buildDirName = name.substring(0, name.indexOf('.')) + "_unzip";
+
+        String reportPrefix = getReportPrefix(itemName);
+        File reportFile = new File(workingDirFile, reportPrefix + ".json");
+        if (reportFile.exists()) {
+            reportFile.delete();
+        }
+        Runtime runtime = Runtime.getRuntime();
+        Process process = null;
+        try {
+            String error = "";
+            FileInputStream input = new FileInputStream(configTemplate);
+            String content = IOUtils.toString(input, CHARSET);
+            input.close();
+
+            String newConfigFileName = itemName + "-config.json";
+            File newConfigFile = new File(workingDirFile, newConfigFileName);
+            if (newConfigFile.exists()) {
+                newConfigFile.delete();
+            }
+
+            FileOutputStream fileOutputStream = new FileOutputStream(newConfigFile);
+            IOUtils.write(
+                    content.replace("@NAME_HOLDER", apk.getAbsolutePath().replace("\\", "\\\\"))
+                            .replace("@REPORT_NAME_HOLDER", new File(workingDirFile, reportPrefix).getAbsolutePath().replace("\\", "\\\\")),
+                    fileOutputStream, CHARSET);
+
+            fileOutputStream.close();
+
+            String command = String.format("java -jar %s --config %s", canaryJar.getAbsoluteFile(), newConfigFile.getAbsolutePath());
+            LOGGER.info(command);
+            process = runtime.exec(command, null, workingDirFile);
+
+            try (InputStream inputStream = process.getInputStream();
+                 InputStream errorStream = process.getErrorStream();
+                 BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream))) {
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
+                    LOGGER.info(line);
+                }
+                code = process.waitFor();
+                error = IOUtils.toString(errorStream, CHARSET);
+            } finally {
+                if (process != null) {
+                    process.destroy();
+                }
+            }
+            FileUtils.deleteDirectory(new File(workingDirFile, buildDirName));
+
+            if (code != 0) {
+                LOGGER.info(error);
+                throw new RuntimeException("failed for " + name);
+            }
+
+            return getApkReportFromJsonReport(reportFile);
+        } catch (IOException | InterruptedException e) {
+            LOGGER.error("error in analyzeApk", e);
+        }
+        return null;
+    }
+
+    public static String getAPKAnalysisItemName(String apkName) {
+        return apkName.replace("-signed-aligned", "").replace(".apk", "");
```

Comment:
What is the purpose of replacing '-signed-aligned' by ''

Reply:
Cleaned. Good callout.